tests/test_tasks.py

```diff
@@ -77,3 +78,118 @@ def test_active_learning_discrete(model, dataset):
+        pass
+    def predict(self, X: np.ndarray) -> np.ndarray:
+        p = np.array([[.1, .2, .7], [.6, .3, .1]])
+        return p
+
+class ApproximateClassificationMock(ClassificationMock):
+    """
+    Approximate classification mock.
+    """
+    def predict(self, X: np.ndarray) -> np.ndarray:
+        p = np.array([[[.1, .2, .7], [.6, .3, .1]], [[.01, .02, .97], [.64, .05, .31]]])
+        return p
+
+class RegressionDataMock(object):
+    """
+    Regression data mock.
+    """
+    X_train, Y_train = np.empty(shape=()), np.empty(shape=())
+    X_test, Y_test = np.empty(shape=()), np.array([[1., 1., 1.], [1., 1., 1.]])
+    Y_std = 2.  # Y_test must be compatible with regression mocks...
+
+class ClassificationDataMock(object):
+    """
+    Classification data mock.
+    """
+    X_train, Y_train = np.empty(shape=()), np.empty(shape=())
+    X_test, Y_test = np.empty(shape=()), np.array([[0], [1]])
+    K = 3  # Y_test and K must be compatible with classification mocks...
+
+# Below correct results computed by hand...
+
+regression_results = {}
+regression_results['test_loglik'] = -9.8963
+regression_results['test_loglik_unnormalized'] = -10.5507
+regression_results['test_mae'] = 2.5
+regression_results['test_mae_unnormalized'] = 5.0
+regression_results['test_rmse'] = 3.0277
+regression_results['test_rmse_unnormalized'] = 6.0553
+
+approximate_regression_results = {}
+approximate_regression_results['test_loglik'] = -10.5197
+approximate_regression_results['test_loglik_unnormalized'] = -11.1836
+approximate_regression_results['test_mae'] = 2.75
+approximate_regression_results['test_mae_unnormalized'] = 5.5
+approximate_regression_results['test_rmse'] = 3.2372
+approximate_regression_results['test_rmse_unnormalized'] = 6.4743
+
+classification_results = {}
+classification_results['test_loglik'] = -1.7533
+classification_results['test_acc'] = 0.0
+classification_results['Y_test'] = np.array([[0], [1]])
+classification_results['p_test'] = np.array([[.1, .2, .7], [.6, .3, .1]])
+
+approximate_classification_results = {}
+approximate_classification_results['test_loglik'] = -2.3217
+approximate_classification_results['test_acc'] = 0.0
+approximate_classification_results['Y_test'] = np.array([[0], [1]])
+approximate_classification_results['p_test'] = np.array([[0.055, 0.11 , 0.835], [0.62 , 0.175, 0.205]])
+
+# Below two tests, one for regression and one for classification (2-dim and 3-dim case respectively)
+
+regression_tuple = (RegressionDataMock(), RegressionMock(), regression_results)
+approx_regression_tuple = (RegressionDataMock(), ApproximateRegressionMock(), approximate_regression_results)
+
+@pytest.mark.parametrize('tuple', [regression_tuple, approx_regression_tuple])
+def test_regression(tuple):
+    data, model, correct_result = tuple
+    result = run_regression(None, data=data, model=model, is_test=True)
+    assert correct_result['test_loglik'] == pytest.approx(result['test_loglik'], 1e-3)
```

Comment:
I think most of us use `np.testing.assert_almost_equal` instead of `pytest.approx`. `np.testing` gives nice and clear errors when the assert isn't fulfilled.